pkg/policy/api/rule.go

```diff
@@ -0,0 +1,234 @@
+	// endpoints which are whitelisted in fromCIDR.
+	//
+	// +optional
+	FromEndpoints []EndpointSelector `json:"fromEndpoints,omitempty"`
+
+	// FromRequires is a selector which must match in combination with at
+	// least one selector defined in FromEndpoints. The selector must match
+	// in any allowed source but by itself does not grant access.  This
+	// allows to define a rule which says, allow "foo" to "bar" but only if
+	// "group=A" is set while "group=A" as a label itself has no immediate
+	// access rights.
+	//
+	// +optional
+	FromRequires []EndpointSelector `json:"fromRequires,omitempty"`
+
+	// ToPorts is a list of destination ports identified by port number and
+	// protocol which the endpoint subject to the rule is allowed to
+	// receive connections on.
+	//
+	// If omitted and no other IngressRule applies a ToPorts rule on the
+	// endpoint, the endpoint is allowed to receive connections on all
+	// ports.
+	//
+	// +optional
+	ToPorts []PortRule `json:"toPorts,omitempty"`
+
+	// FromCIDR is a list of IP blocks which the endpoint subject to the
+	// rule is allowed to receive connections from. This will match on the
+	// source IP address of incoming connections.
+	//
+	// This restriction does not apply to connections from endpoints which
+	// have been whitelisted using a FromEndpoints rule of an IngressRule.
+	//
+	// If omitted and no other IngressRule applies a fromCIDR rule on the
+	// endpoint, the endpoint is only allowed to take connections from
+	// endpoints which are whitelisted in FromEndpoints.
+	//
+	// +optional
+	FromCIDR []CIDR `json:"toCIDR,omitempty"`
+}
+
+// EgressRule contains all rule types which can be applied at egress, i.e.
+// network traffic that originates inside the endpoint and exits the endpoint
+// selected by the endpointSelector.
+//
+type EgressRule struct {
+	// ToPorts is a list of destination ports identified by port number and
+	// protocol which the endpoint subject to the rule is allowed to
+	// connect to.
+	//
+	// If omitted and no other IngressRule applies a ToPorts rule on the
+	// endpoint, the endpoint is allowed to connect on all ports.
+	//
+	// +optional
+	ToPorts []PortRule `json:"toPorts,omitempty"`
+
+	// ToCIDR is a list of IP blocks which the endpoint subject to the rule
+	// is allowed to initiate connections to. This restriction does not
+	// apply to connections to endpoints which have been whitelisted using
+	// a FromEndpoints rule of an IngressRule.
+	//
+	// This will match on the destination IP address of outgoing
+	// connections.
+	//
+	// If omitted and no other EgressRule applies a ToCIDR rule on the
+	// endpoint, the endpoint is allowed to make outgoing connections to
+	// any IP address.
+	//
+	// +optional
+	ToCIDR []CIDR `json:"toCIDR,omitted"`
+}
+
+// CIDR TODO
+type CIDR struct {
+	foo string
+}
+
+// PortProtocol specifices an L4 port with an optional protocol
+//
+// TODO: allow port range?
```

Comment:
It's usually useful. Actually, a list combining individual ports and ranges seems common, e.g.:
["80", "443", "2000-2100"]
The only negative aspect about it is that it requires the data to be passed as strings, but that's a small price to pay.